### PR TITLE
App 3230

### PR DESCRIPTION
**Pull request by Werder630**

### CODE REVIEW

File: src/main/java/com/yandex/money/api/model/Operation.java [hunk 1]
```diff
@@ -200,10 +214,6 @@ public boolean isRepeatable() {
         return repeatable != null && repeatable;
     }
 
-    public boolean isFavorite() {
```
Comment (yasevich) [inline src/main/java/com/yandex/money/api/model/Operation.java]: Unnecessary change.

File: src/main/java/com/yandex/money/api/model/Operation.java [hunk 2]
```diff
@@ -190,6 +202,8 @@ protected Operation(Builder builder) {
         paymentParameters = Collections.unmodifiableMap(checkNotNull(builder.paymentParameters,"paymentParameters"));
         favorite = builder.favorite;
         digitalGoods = builder.digitalGoods;
+        categories = Collections.unmodifiableList(builder.categories);
```
Comment (yasevich) [inline src/main/java/com/yandex/money/api/model/Operation.java]: No consistency check for `build.categories`.

File: src/main/java/com/yandex/money/api/typeadapters/model/OperationTypeAdapter.java [hunk 3]
```diff
@@ -122,6 +129,8 @@ public Operation deserialize(JsonElement json, Type typeOfT, JsonDeserialization
                     .setFavorite(getBoolean(o, MEMBER_FAVOURITE))
                     .setDigitalGoods(DigitalGoodsTypeAdapter.getInstance().fromJson(o.get(
                             MEMBER_DIGITAL_GOODS)))
+                    .setCategories(new Gson().fromJson(o.getAsJsonArray(MEMBER_CATEGORIES), listType))
```
Comment (yasevich) [inline src/main/java/com/yandex/money/api/typeadapters/model/OperationTypeAdapter.java]: Use `GsonProvider` to get `Gson` instance.

File: src/main/java/com/yandex/money/api/typeadapters/model/OperationTypeAdapter.java [hunk 4]
```diff
@@ -167,6 +176,12 @@ public JsonElement serialize(Operation src, Type typeOfSrc, JsonSerializationCon
         if (src.digitalGoods != null) {
             object.add(MEMBER_DIGITAL_GOODS, DigitalGoodsTypeAdapter.getInstance().toJsonTree(src.digitalGoods));
         }
+        if (!(src.categories == null || src.categories.isEmpty())) {
```
Comment (yasevich) [inline src/main/java/com/yandex/money/api/typeadapters/model/OperationTypeAdapter.java]: Judging by the code above `Operation`, categories can never be `null`.

File: src/main/java/com/yandex/money/api/model/Operation.java (hunk 1 above)
Comment (Werder630) [inline src/main/java/com/yandex/money/api/model/Operation.java]: Yes, revert

File: src/main/java/com/yandex/money/api/typeadapters/model/OperationTypeAdapter.java (hunk 3 above)
Comment (Werder630) [inline src/main/java/com/yandex/money/api/typeadapters/model/OperationTypeAdapter.java]: Done

File: src/main/java/com/yandex/money/api/typeadapters/model/OperationTypeAdapter.java (hunk 4 above)
Comment (Werder630) [inline src/main/java/com/yandex/money/api/typeadapters/model/OperationTypeAdapter.java]: Done

File: src/main/java/com/yandex/money/api/model/Operation.java (hunk 2 above)
Comment (Werder630) [inline src/main/java/com/yandex/money/api/model/Operation.java]: Done

File: src/main/java/com/yandex/money/api/typeadapters/model/OperationTypeAdapter.java [hunk 5]
```diff
@@ -167,6 +177,12 @@ public JsonElement serialize(Operation src, Type typeOfSrc, JsonSerializationCon
         if (src.digitalGoods != null) {
             object.add(MEMBER_DIGITAL_GOODS, DigitalGoodsTypeAdapter.getInstance().toJsonTree(src.digitalGoods));
         }
+        if (!src.categories.isEmpty()) {
+            object.add(MEMBER_CATEGORIES, new Gson().toJsonTree(src.categories));
```
Comment (yasevich) [inline src/main/java/com/yandex/money/api/typeadapters/model/OperationTypeAdapter.java]: Use `GsonProvider`.

Comment (Werder630) [inline src/main/java/com/yandex/money/api/typeadapters/model/OperationTypeAdapter.java]: Done
